pkg/controller/eks/addon/setup.go

```diff
@@ -144,10 +149,69 @@ func (h *hooks) isUpToDate(cr *eksv1alpha1.Addon, resp *awseks.DescribeAddonOutp
 		return false, nil
 	}
 
+	// AddOn Configuration Values
+	configUpToDate, configUpToDateErr := isUpToDateConfigurationValues(cr, resp)
+	if configUpToDateErr != nil {
+		return false, configUpToDateErr
+	}
+	if !configUpToDate {
+		return false, nil
+	}
+
 	add, remove := awsclients.DiffTagsMapPtr(cr.Spec.ForProvider.Tags, resp.Addon.Tags)
 	return len(add) == 0 && len(remove) == 0, nil
 }
 
+func isUpToDateConfigurationValues(cr *eksv1alpha1.Addon, obj *awseks.DescribeAddonOutput) (bool, error) {
+	// Handle nil pointer refs
+	crConfigurationValues := aws.StringValue(cr.Spec.ForProvider.ConfigurationValues)
+	objConfigurationValues := aws.StringValue(obj.Addon.ConfigurationValues)
+	if crConfigurationValues == "" && objConfigurationValues == "" {
+		return true, nil
+	}
+
+	if crConfigurationValues == "" && objConfigurationValues != "" {
+		return false, nil
+	}
+	if crConfigurationValues != "" && objConfigurationValues == "" {
+		return false, nil
+	}
+
+	// Normalize the data
+	objConfigurationValues, objErr := convertConfigurationValuesToString(objConfigurationValues)
+	if objErr != nil {
+		return false, objErr
+	}
+	crConfigurationValues, crErr := convertConfigurationValuesToString(crConfigurationValues)
+	if crErr != nil {
+		return false, crErr
+	}
+	// Compare normalized strings
+	return cmp.Equal(objConfigurationValues, crConfigurationValues), nil
+}
+
+// convertConfigurationValuesToString will deserialize and serialize in order to normalize
+func convertConfigurationValuesToString(configurationValues string) (string, error) {
+	var jsonConfigurationValues map[string]interface{}
+	if strings.HasPrefix(configurationValues, "{") {
```

Comment:
That doesn't work if the JSON is not starting with `{`. It would be better to just use `json.Unmarshal` and directly and use `yaml.Unmarshal` if that fails.

Reply:
I initially used the return error - if `SyntaxError` then assume it's yaml.  I didn't like that approach as it would try to assume yaml even if the json syntax was incorrect.
There is a json schema published which should enforce that the json MUST start with `{` so IMO it was safer to do it this way.   Also....the json is much more difficult to use as you have to minimize or escape it and as most k8s is defined in yaml.  I will try yaml parsing first.